src/diffusers/schedulers/scheduling_lcm_flax.py

```diff
@@ -0,0 +1,467 @@
+            raise ValueError(
+                "Number of inference steps is 'None', you need to run 'set_timesteps' after creating the scheduler"
+            )
+        
+        step_index = jax.lax.select(state.step_index == -1, self._init_step_index(state, timestep), state.step_index)
+        
+        # 1. get previous step value        
+        prev_step_index = step_index + 1
+        prev_timestep = jax.lax.select(prev_step_index < len(state.timesteps), state.timesteps[prev_step_index], timestep)
+        jax.debug.print("prev_timestep: {x}", x=prev_timestep)
+        # 2. compute alphas, betas
+        jax.debug.print("timestep: {x}",x=timestep)
+        alpha_prod_t = state.common.alphas_cumprod[timestep]
+        jax.debug.print("alpha_prod_t: {x}",x=alpha_prod_t)
+        alpha_prod_t_prev = jax.lax.select(prev_timestep >=0, state.common.alphas_cumprod[prev_timestep], state.final_alpha_cumprod)
+        
+        beta_prod_t = 1 - alpha_prod_t
+        beta_prod_t_prev = 1 - alpha_prod_t_prev
+
+        # 3. Get scalings for boundary conditions
+        c_skip, c_out = self.get_scalings_for_boundary_condition_discrete(timestep)
+
+        # 4. Compute the predicted original sample x_0 based on the model parameterization
+        if self.config.prediction_type == "epsilon":  # noise-prediction
+            predicted_original_sample = (sample - jnp.sqrt(beta_prod_t) * model_output) / jnp.sqrt(alpha_prod_t)
+        elif self.config.prediction_type == "sample":  # x-prediction
+            predicted_original_sample = model_output
+        elif self.config.prediction_type == "v_prediction":  # v-prediction
+            predicted_original_sample = jnp.sqrt(alpha_prod_t) * sample - jnp.sqrt(beta_prod_t.sqrt) * model_output
+        else:
+            raise ValueError(
+                f"prediction_type given as {self.config.prediction_type} must be one of `epsilon`, `sample` or `v_prediction` for `LCMScheduler`."
+            )
+
+        # 5. Clip or threshold "predicted x_0"
+        # if self.config.thresholding:
+        #     predicted_original_sample = self._threshold_sample(predicted_original_sample)
+        # elif self.config.clip_sample:
+        #     predicted_original_sample = predicted_original_sample.clamp(
+        #         -self.config.clip_sample_range, self.config.clip_sample_range
+        #     )
+
+        # 6. Denoise model output using boundary conditions
+        denoised = c_out * predicted_original_sample + c_skip * sample
+
+        # 7. Sample and inject noise z ~ N(0, I) for MultiStep Inference
+        # Noise is not used on the final timestep of the timestep schedule.
+        # This also means that noise is not used for one-step sampling.
+
+        jax.debug.print("**step_index: {x}",x=step_index)
+        jax.debug.print("**state.step_index: {x}",x=state.step_index)
+
+        def get_noise(key, shape, dtype):
+            jax.debug.print("---------------get_noise")
```

Comment:
I wonder if we should split the key here and store it as part of the `state`. As commented in https://github.com/entrpn/diffusers/pull/1, I believe that just splitting, [like `FlaxDDPMScheduler` does](https://github.com/huggingface/diffusers/blob/774f5c45817805546ae5eb914c175d4fe72dcfe9/src/diffusers/schedulers/scheduling_ddpm_flax.py#L267), is not correct.